libraries/SuplaDevice/supla/network/esp_wifi.h

```diff
@@ -68,24 +74,60 @@ class ESPWifi : public Supla::Network {
     }
     Serial.println("]");
 #endif
-    long sendSize = client.write((const uint8_t *)buf, count);
+    long sendSize = client->write((const uint8_t *)buf, count);
     return sendSize;
   }
 
   bool connect(const char *server, int port) {
-    return client.connect(server, port);
+    String message;
+    if (client == NULL) {
+      if (isSecured) {
+        message = "secured connection";
+        client = new WiFiClientSecure();
+        if (fingerprint.length() > 0) {
+          message += " with certificate matching";
+          ((WiFiClientSecure *)client)->setFingerprint(fingerprint.c_str());
+        } else {
+          message += " without certificate matching";
+          ((WiFiClientSecure *)client)->setInsecure();
+        }
+      } else {
+        message = "unsecured connection";
+        client = new WiFiClient();
+      }
+    }
+
+    supla_log(LOG_DEBUG,
+              "Establishing %s with: %s (port: %d)",
+              message.c_str(),
+              server,
+              this->port);
+
+    bool result = client->connect(server, this->port);
+
+    if (result && isSecured) {
+      if (!((WiFiClientSecure *)client)->verify(fingerprint.c_str(), server)) {
+        supla_log(LOG_DEBUG, "Provided certificates doesn't match!");
+        client->stop();
+        return false;
+      }
+    };
+
+    return result;
   }
 
   bool connected() {
-    return client.connected();
+    return (client != NULL) && client->connected();
```

Comment:
just:
return client && client->connected();
will be enough
Or at least use nullptr instead of NULL